dataservice/api/phenotype/resources.py

```diff
@@ -0,0 +1,162 @@
+from flask import abort, request
+from sqlalchemy.orm.exc import NoResultFound
+from marshmallow import ValidationError
+
+from dataservice.extensions import db
+from dataservice.api.common.pagination import paginated, Pagination
+from dataservice.api.phenotype.models import Phenotype
+from dataservice.api.phenotype.schemas import PhenotypeSchema
+from dataservice.api.common.views import CRUDView
+
+
+class PhenotypeListAPI(CRUDView):
+    """
+    Phenotype REST API
+    """
+    endpoint = 'phenotypes_list'
+    rule = '/phenotypes'
+    schemas = {'Phenotype': PhenotypeSchema}
+
+    @paginated
+    def get(self, after, limit):
+        """
+        Get all phenotypes
+        ---
+        description: Get all phenotypes
+        template:
+          path:
+            get_list.yml
+          properties:
+            resource:
+              Phenotype
+        """
+        q = Phenotype.query
+
+        return (PhenotypeSchema(many=True)
+                .jsonify(Pagination(q, after, limit)))
+
+    def post(self):
+        """
+        Create a new phenotype
+        ---
+        template:
+          path:
+            new_resource.yml
+          properties:
+            resource:
+              Phenotype
+        """
+
+        body = request.json
+
+        # Deserialize
+        try:
+            p = PhenotypeSchema(strict=True).load(body).data
+        # Request body not valid
+        except ValidationError as e:
+            abort(400, 'could not create phenotype: {}'.format(e.messages))
+
+        # Add to and save in database
+        db.session.add(p)
+        db.session.commit()
+
+        return PhenotypeSchema(201, 'phenotype {} created'
+                               .format(p.kf_id)).jsonify(p), 201
+
+
+class PhenotypeAPI(CRUDView):
+    """
+    Phenotype REST API
+    """
+    endpoint = 'phenotypes'
+    rule = '/phenotypes/<string:kf_id>'
+    schemas = {'Phenotype': PhenotypeSchema}
+
+    def get(self, kf_id):
+        """
+        Get a phenotype by id
+        ---
+        template:
+          path:
+            get_by_id.yml
+          properties:
+            resource:
+              Phenotype
+        """
+        # Get one
+        try:
+            p = Phenotype.query.filter_by(kf_id=kf_id).one()
```

Comment:
Let's start using get() instead of filter_by now that kf_id is the primary key. See #173 for example